### PR TITLE
Display the correct line number on existing comments

### PR DESCRIPTION
**Pull request by lynn-kim**
## Notion ticket link

[Display the correct line number on existing comments](https://www.notion.so/uwblueprintexecs/Display-the-correct-line-number-on-existing-comments-4a8316cf49bd4c6fa8fa488e457f2991)

## Implementation description

- Fetch the line number from the StoryTranslationContent in backend, 
- *I think there was some confusion around the naming with regards to comment index which led to the bugs*, renamed them to hopefully be clearer (?):
  - `commentIndex` and `WIPCommentIndex` refer to the comment number for a specific thread (the first comment in any thread will have commentIndex=0, the first reply to that comment will be commentIndex=1)
  - `lineIndex` is the specific StoryTranslationContent line number that the comment refers to 
  
Screen Shot 2021-11-16 at 10 14 48 PM

## Steps to test

1. Log in as carl sagan and go to http://localhost:3000/translation/6/13 
2. Check that the comment indices are correct

## What should reviewers focus on?

- Does it work? 
- Does the naming make sense?

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by gaoxk — CHANGES_REQUESTED** [review]
found a bug:
- click big blue comment button
- click comment button by line 2
- make a comment
- save
- existing comment shows up as line 1
- refresh page
- existing comment shows up as line 2 (correct)

File: frontend/src/APIClients/queries/CommentQueries.ts [hunk 1]
```diff
@@ -17,6 +18,7 @@ export type CommentResponse = {
   time: string;
   resolved: boolean;
   content: string;
+  lineIndex?: number;
```
[the remark this answers is not in the document]
Comment (lynn-kim) [reply frontend/src/APIClients/queries/CommentQueries.ts]: Removed!

File: frontend/src/components/review/CommentsPanel.tsx [hunk 2]
```diff
@@ -76,11 +76,10 @@ const CommentsPanel = ({
               resolved={comment.resolved}
               content={comment.content}
               time={comment.time}
+              lineIndex={comment.lineIndex || 0}
```
[the remark this answers is not in the document]
Comment (lynn-kim) [reply frontend/src/components/review/CommentsPanel.tsx]: Fixed here https://github.com/uwblueprint/planet-read/pull/162/commits/45848693a84a6bbe495448dd58cca1129f08f31e

**Review by gaoxk — APPROVED** [review]
noiceee 
just an fyi WIPComment line # is a little funky. going to slot into a different ticket
image

**Pull request merged**